### PR TITLE
Arachnid species

### PR DESCRIPTION
**Pull request by PixelTheKermit**
## About the PR

Adds a new arachnid species. Right now they are a bit faster than a regular human, have an "animal" stomach, take less barotrauma damage, and have 2 extra "pocket" slots at the expense of no outer storage, being more vulnerable to blunt and pierce damage, oxygen takes longer to recover + being more vulnerable to asphyxiation, and they don't do that well to the cold. They also metabolise foods and fluids slower, which can be both a good and bad thing depending on the situation.

_TODO:_
- [x] Make them somewhat unique (but not make them essentially a separate game)
- [x] Work on adding more markings
- [ ] More organ sprites?
- [x] Add more media for the PR
- [x] Spider voices for screaming, speaking + other (?)

~~_**Any ideas for them would be appreciated, and the more people helping the more better they will be.**_~~ this was a mistake

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

screams

image

image

https://user-images.githubusercontent.com/85175107/219976011-c47b3403-0681-4398-af6a-cc4e7058a362.mp4

**Changelog**

:cl: PixelTK
- add: Nanotrasen has experimented with hiring some arachnid-like humanoids.

### CODE REVIEW

**Comment by PixelTheKermit** [comment]
Again the more people helping, the better I can make this

**Comment by 0x6273** [comment]
> , and cannot wear anything externally

So they can't wear hardsuits? How would you do salv or nukeops as an arachnid?

**Comment by PixelTheKermit** [comment]
> > , and cannot wear anything externally
> 
> So they can't wear hardsuits? How would you do salv or nukeops as an arachnid?

Well the reduced barotrauma damage is supposed to help, that combined with any chem that heats you up and a good supply of O2 and for the most part, you're fine

**Comment by PixelTheKermit** [comment]
Doesn't matter now I reverted the "no hardsuit" change

**Comment by PixelTheKermit** [comment]
Okay so I decided to see if adding extra "pocket slots" would be good. These don't require a jumpsuit yet their contents can be seen by anyone
image

image

Again, any opinions on this would be useful.

**Comment by Cheackraze** [comment]
I'm seeing only buffs for this race, they need to be decidedly not better than base humans or they will be power gamed hard

**Comment by PixelTheKermit** [comment]
> I'm seeing only buffs for this race, they need to be decidedly not better than base humans or they will be power gamed hard

Yeah I'm planning to give them somewhat significant nerfs

**Comment by Coopsbomb4** [comment]
I'm going to put my opinion here, revert the removal of no hardsuits. Its a major debuff sure but provides plenty of interesting choices. You also don't get cucked out of roles due to the decreased pressure damage

**Comment by PixelTheKermit** [comment]
Brought back no hardsuits, lets see how bad this goes

**Comment by metalgearsloth** [comment]
Unless you intend to resprite every piece of clothing this isn't going to work because the silhouette doesn't match.

**Comment by PixelTheKermit** [comment]
> Unless you intend to resprite every piece of clothing this isn't going to work because the silhouette doesn't match.

I'm going to think about this tomorrow, as right now it's time for me to sleep...
...fuck

**Comment by PixelTheKermit** [comment]
Okay so I'm going to assume it's better for me to just resprite the arachnid sprite to work better with clothing. Not sure if that's the best choice but the alternative is spriting all the clothing to work so I likely don't have a choice.

**Comment by PixelTheKermit** [comment]
> Unless you intend to resprite every piece of clothing this isn't going to work because the silhouette doesn't match.

I have decided to resprite the arachnid to better fit clothing

**Comment by PixelTheKermit** [comment]
I changed the two "limb pockets" into "web pockets". This makes more sense (somewhat), however I may need to add more sprites to show these "web pockets", _sigh_ the pain of creating anything

**Comment by PixelTheKermit** [comment]
Might need to buff barotrauma damage

**Comment by arealbeba** [comment]
considering they can't wear a hardsuit it's fine

**Comment by PixelTheKermit** [comment]
I'm almost done with arachnids! Just need to make more sprites for markings, organs and... I think that's all! Unless there's something else I'm missing

**Review by 0x6273 — CHANGES_REQUESTED** [review]
You should use spaces in the .json files instead of tabs. They now have a mix of both spaces and tabs.

Also, some .ftl and .yml files are missing line breaks:
image

The `.editorconfig` forbids both tab indentation and missing line breaks. Fix your editor so it respects the editorconfig settings.

**Comment by PixelTheKermit** [comment]
> You should use spaces in the .json files instead of tabs. They now have a mix of both spaces and tabs.
> 
> Also, some .ftl and .yml files are missing line breaks: image
> 
> The `.editorconfig` forbids both tab indentation and missing line breaks. Fix your editor so it respects the editorconfig settings.

Think I fixed both issues. Also found a plugin for notepad++ for the last bit, thanks for letting me know about doing that

**Comment by PixelTheKermit** [comment]
Since I'm not changing this anymore, I'm considering opening this for merging.

**Comment by PixelTheKermit** [comment]
Ready for review, any criticism would be appreciated now.

**Comment by PixelTheKermit** [comment]
Still waiting for judgement day.

**Comment by Peptide90** [comment]
If I’m reading right, you have barotrauma at 0.15 now compared to 0.35? I get they have an exoskeleton so that’s fine but I am hesitant. As long as they still take damage.

Also minor point but arachnids aren’t insects. You have a comment on their heat resistance about it, which yeah they like body heat but the whole kill it with fire meme should rain true I feel, so fire should ruin them.

**Comment by PixelTheKermit** [comment]
> Also minor point but arachnids aren’t insects. You have a comment on their heat resistance about it, which yeah they like body heat but the whole kill it with fire meme should rain true I feel, so fire should ruin them.

I'll experiment with more heat damage.

**Comment by PixelTheKermit** [comment]
Updated the default marking to look slightly better

screams

**Comment by PixelTheKermit** [comment]
and I somehow messed up the json give me a minute

**Comment by Froozigiusz** [comment]
> More organ sprites?
 
 I somehow read this the first time as a "Male organ sprites" :skull:

**Comment by PixelTheKermit** [comment]
One C# edit and tests fail ffs

**Comment by PixelTheKermit** [comment]
Okay I'm 90% sure I cannot fix the failing tests without removing the blue blood.

**Comment by deltanedas** [comment]
test failure is unrelated, happening on other prs too

**Comment by PixelTheKermit** [comment]
> test failure is unrelated, happening on other prs too

Ah understood

File: Resources/Audio/Voice/Arachnid/attributions.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+- files: ["arachnid_scream.ogg"]
+  license: "CC-BY-4.0"
+  copyright: "Recorded by PixelTK"
+  source: "https://github.com/space-wizards/space-station-14/pull/13945"
+
+- files: ["arachnid_laugh.ogg"]
+  license: "CC-BY-4.0"
+  copyright: "Recorded by PixelTK"
+  source: "https://github.com/space-wizards/space-station-14/pull/13945"
```
Comment (metalgearsloth) [inline Resources/Audio/Voice/Arachnid/attributions.yml:9]: Just put them in the same files list.

File: Resources/Audio/Voice/Talk/license.txt [hunk 2]
```diff
@@ -23,3 +23,7 @@ buwoo_exclaim (Renamed to slime_exclaim.ogg)
 all taken from
 https://github.com/goonstation/goonstation/tree/eb3e7df6292d23f6af2f18b4372d3a8ba4b0fda7/sound/misc/talk
 licensed under CC BY-NC-SA 3.0
+arachnid.ogg
+arachnid_ask.ogg
+arachnid_exclaim.ogg
+arachnid voices by PixelTK licensed under CC-BY-4.0
```
Comment (metalgearsloth) [inline Resources/Audio/Voice/Talk/license.txt:29]: Needs to be in attributions.yml

File: Resources/Prototypes/Damage/modifier_sets.yml [hunk 3]
```diff
@@ -118,6 +118,16 @@
     Heat: 1.5
     Shock: 1.2
 
+- type: damageModifierSet
+  id: Arachnid # Don't do too well with high temperatures, venomous (well some kinds anyways) and have an exo-skeleton (so probably harder to stab but easier to... break?)
+  coefficients:
+    Blunt: 1.35
+    Piercing: 1.2
+    Slash: 0.85
+    Cold: 1.15
```
Comment (metalgearsloth) [inline Resources/Prototypes/Damage/modifier_sets.yml:127]: How come blunt piercing and cold are different?

File: Resources/Textures/Mobs/Species/Arachnid/organs.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      "name": "eyeball-l"
+    },
+	{
+      "name": "eyeball-r"
+    },
+	{
+      "name": "tongue"
+    }
+  ]
+}
```
Comment (metalgearsloth) [inline Resources/Textures/Mobs/Species/Arachnid/organs.rsi/meta.json:20]: Formatting needs fixing and copyright needs something more like a github user or discord handle.

File: Resources/Textures/Mobs/Species/Arachnid/parts.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    {
+      "name": "eyes",
+      "directions": 4
+    },
+    {
+	  "name": "template",
+	  "directions": 4
+    }
+  ]
+}
```
Comment (metalgearsloth) [inline Resources/Textures/Mobs/Species/Arachnid/parts.rsi/meta.json:70]: Same here.

File: Resources/Textures/Mobs/Customization/arachnid.rsi/meta.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      "name": "sharp",
+      "directions": 4
+    },
+    {
+      "name": "web",
+      "directions": 4
+    },
+	{
+      "name": "stripes",
+      "directions": 4
```
Comment (metalgearsloth) [inline Resources/Textures/Mobs/Customization/arachnid.rsi/meta.json:24]: Here

File: Resources/Textures/Interface/Default/meta.json [hunk 7]
```diff
@@ -81,6 +81,9 @@
       },
       {
         "name": "pda"
+      },
+	  {
+        "name": "web"
       }
```
Comment (metalgearsloth) [inline Resources/Textures/Interface/Default/meta.json:87]: .

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by metalgearsloth** [comment]
Is there a full list of what's changed somewhere.

**Comment by PixelTheKermit** [comment]
working on resolving the issues right now

File: Resources/Audio/Voice/Talk/license.txt (hunk 2 above)
Comment (PixelTheKermit) [reply Resources/Audio/Voice/Talk/license.txt:29]: There doesn't seem to be a attributions.yml in Resources\Audio\Voice\Talk, should I just make one and move all the stuff I did there?

Comment (metalgearsloth) [reply Resources/Audio/Voice/Talk/license.txt]: Yeah

Comment (PixelTheKermit) [reply Resources/Audio/Voice/Talk/license.txt]: right did that.

**Review by metalgearsloth — APPROVED** [review]

**Comment by VasilisThePikachu** [comment]
Wait isn't there a species freeze currently?

**Comment by metalgearsloth** [comment]
> Wait isn't there a species freeze currently?

Was added after this PR.

**Pull request merged**